### DISCUSSION TITLE
Layer shell, keyboard interactivity, and tooltip

### DISCUSSION THREAD

**Discussion by ATM-Jahid** [hyprwm/Hyprland#4308] [Q&A]
I am trying to build a panel using layer-shell-qt. So far, I have a [small prototype](https://github.com/ATM-Jahid/qtventure/blob/main/proto/main.cpp). The program reads the contents of "hyprland clients" and shows the client class icons. The client titles are displayed with tooltips. 

The problem is the tooltips don't get displayed when keyboard interactivity is set to "none". Tooltips are visible when the interactivity is set to "exclusive/on-demand". I don't know why the level of keyboard interactivity should change the behavior of the tooltips. I would appreciate any suggestions on how to solve this.

Other observations:

- Waybar sets keyboard interactivity to "none" and its tooltips work.
- My prototype panel receives mouse events (click, hover) even when interactivity is set to "none". Only the tooltips don't work.

**Comment by vaxerski** [comment]
that's... odd.
Does it work on sway?

**Reply by ATM-Jahid** [reply]
I just did some testing on sway and wayfire. No tooltips are shown on both compositors if interactivity is set to none. However, both show the tooltips when interactivity is "exclusive/on-demand" (at the expense of being unable to focus on other windows). 

Surprisingly, Hyprland does show the tooltips when interactivity is set to "none" but only when there are no other clients in the workspace. (I noticed this recently.) Hyprland is different from sway/wayfire in that respect.

I have updated my prototype (link in the post) to include a QPushButton to do the testing on other compositors. I can also provide a minimal reproducible example if needed.
